app/core/Engine.ts

```diff
@@ -1321,6 +1334,22 @@ class Engine {
       },
     );
 
+    this.controllerMessenger.subscribe(
```

Comment:
Hmm, this may still leave us in an invalid state _shortly after_ a switch, or if we switch to a network that fails to respond correctly to `net_version`.

Perhaps we should clear the `networkId` state when a switch begins, so that it's not set to the wrong value during the switch, or after a failure. e.g.

```suggestion
    this.controllerMessenger.subscribe(
      'NetworkController:networkWillChange',
      () => {
        store.dispatch(setNetworkId(''));
      }
    );

    this.controllerMessenger.subscribe(
```

Reply:
Though it may be more aligned with Redux best practices to create a new action + action creator for this case (e.g. `networkChanged` and `NETWORK_CHANGED`) rather than re-using `setNetworkId` this way.

Reply:
Ahhh yes!!! This will make it bullet proof!